### PR TITLE
[gosrc2cpg]: Call node for struct declaration

### PR DESCRIPTION
**Pull request by hiteshbedre**
For structure declaration statement created a call node with structure parameter as callnode's argument.
Below are different ways we can declare a structure in GO:

`rect := Rectangle{width: 10, height: "somestr"}`
`rect := Rectangle{}`
`rect := somepackage.Rectangle{l: 10, w: 20}`
`
square := Square{length: 10, width: 20}
rect := Rectangle{sq: square}
`
`var square = Square{length: 10}`
`var square = Square{10}`
`var rectangle = Rectangle{10, 20}`

### CODE REVIEW

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        |func main() {
+        |    rect := Rectangle{width: 10, height: "somestr"}
+        |}
+        |""".stripMargin,
+      "test.go"
+    )
+    val List(rectangleCallNode) = cpg.call.nameExact("Rectangle").l
+    "test basic cpg properties for call node of structure declaration" in {
+      rectangleCallNode.name shouldBe "Rectangle"
+      rectangleCallNode.methodFullName shouldBe "main.Rectangle"
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala:23]: This looks like a constructor to me in some sense, so should we not stick to the convention of calling the constructor method `<init>`?

Comment (hiteshbedre) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala:23]: Addressed here: https://github.com/joernio/joern/pull/3596/commits/cb42da630eaa582260c9341c5b345b16895233c7
Added postfix ".<init>" for every structure declaration

File: joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        |    rect := somepackage.Rectangle{l: 10, w: 20}
+        |}
+        |""".stripMargin,
+      "test.go"
+    )
+    val List(rectangleCallNode) = cpg.call.nameExact("Rectangle").l
+    "test basic cpg properties for call node of structure declaration" in {
+      rectangleCallNode.name shouldBe "Rectangle"
+      rectangleCallNode.methodFullName shouldBe "path/to/somepackage.Rectangle"
+      rectangleCallNode.typeFullName shouldBe "path/to/somepackage.Rectangle.<init>"
```
Comment (DavidBakerEffendi) [inline joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala:74]: The constructor is not a type, and rather the `methodFullName` should be `path/to/somepackage.Rectangle.<init>`

Comment (hiteshbedre) [reply joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/StructureCallTests.scala]: Thanks for pointing it out.
Corrected here: https://github.com/joernio/joern/pull/3596/commits/4d71b6e52c2b74cc54001de2a5eb74427307d804

**Review by DavidBakerEffendi — APPROVED** [review]

**Pull request merged**